client/src/components/geneExpression/menus/editGenesetNameDialogue.js

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import { connect } from "react-redux";
+import AnnoDialog from "../../categorical/annoDialog";
+import LabelInput from "../../categorical/labelInput";
+
+@connect((state) => ({
+  annotations: state.annotations,
+  schema: state.annoMatrix?.schema,
+  ontology: state.ontology,
+  obsCrossfilter: state.obsCrossfilter,
+  genesetsUI: state.genesetsUI,
+}))
+class RenameGeneset extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      genesetName: "",
+    };
+  }
+
+  disableEditGenesetNameMode = (e) => {
+    const { dispatch } = this.props;
+    this.setState({
+      genesetName: "",
+    });
+    dispatch({
+      type: "geneset: disable rename geneset mode",
+    });
+    if (e) e.preventDefault();
+  };
+
+  renameGeneset = (e) => {
+    const { dispatch, genesetsUI } = this.props;
+    const { genesetName } = this.state;
+
+    dispatch({
+      type: "geneset: rename",
+      name: genesetsUI.isEditingGenesetName,
+      newName: genesetName,
+    });
+    dispatch({
+      type: "geneset: disable rename geneset mode",
+    });
+    e.preventDefault();
+  };
+
+  genesetNameError = () => {
+    /* todo genesets validation */
```

Comment:
is this TODO captured in an issue, or should it be in this PR?

Reply:
Capturing in issue.